client/my-sites/theme/main.jsx

```diff
@@ -130,10 +130,11 @@ const ThemeSheet = React.createClass( {
 	},
 
 	getValidSections() {
+		const { isCurrentUserPaid, isJetpack, forumUrl, isWpcomTheme, isLoggedIn } = this.props;
 		const validSections = [];
 		validSections.push( '' ); // Default section
 		this.props.supportDocumentation && validSections.push( 'setup' );
-		validSections.push( 'support' );
+		( ! isLoggedIn || ( isCurrentUserPaid && ! isJetpack ) || forumUrl || isWpcomTheme ) && validSections.push( 'support' );
```

Comment:
Yes that's the reason why I'm a bit uncomfortable with that.

UX wise it's better overall to hide it — but ok, let me thing as a backup solution if I can come up with a message.